spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -252,7 +104,7 @@ public synchronized Object invoke(MethodInvocation invocation) throws Throwable
 	@Override
 	public synchronized Object getObject() throws Exception {
 		if (this.proxy == null) {
-			createChannels(this.type);
+			this.channelFactory.createChannels(this.type);
```

Comment:
this feels like a brittle contract... having to call `createChannels` prior to calling `getInputs` or `getOutputs`